sonic_platform_base/module_base.py

```diff
@@ -462,3 +466,33 @@ def is_midplane_reachable(self):
             A bool value, should return True if module is reachable via midplane
         """
         return NotImplementedError
+
+    ##############################################
+    # Asic methods
+    ##############################################
+    def get_all_asics(self):
+        """
+        Retrieves the list of all asics on the module that are visible in PCI domain.
+        When called from the Supervisor of modular system, the module could be
+        fabric card, and the function returns all fabric asics on this module that
+        appear in PCI domain of the Supervisor.
+
+        Returns:
+            A list of asics. Index of an asic in the list is the index of the asic
+            on the module. Index is 0 based.
+
+            An item in the list is a tuple that includes:
+               - Asic instance number (indexed globally across all modules of
```

Comment:
Looks ok to me. @ngoc-do the only question I have is regarding the Asic instance number, does it need to be unique number valid globally across the chassis ? or can it be a number locally significant (0,1,2 for a 3 ASIC module). We can derive the globally significant asic_num = slot_num * locally significant asic_num. ?

Reply:
Yes, it needs to be unique across the chassis. It will be asic0, asic1, asic2, asic3, asic4... asic11 in a system of 12 asics (6 cards, 2 chips per card).

Yes, the can get asic_num from the globally significant asic_num = slot_num * locally significant asic_num (or whatever a vendor wants).

Reply:
I am thinking of a scenario where we swap a module with for eg: 2 ASIC's with another module with say 3 ASIC's. The global asic indices which we stored in CHASSIS_ASIC_TABLE needs to be updated ? as now we have a new module with more ASICs

That was the reason why I thought, it is good for every module just gives back the asic id which is local eg: [('0', '0000:05:00.0'), ('1', '0000:07:00.0')] and the key for CHASSIS_ASIC_TABLE could be something like card_index as below.

127.0.0.1:6379[6]> "HGETALL CHASSIS_ASIC_TABLE| card-index"
1) "asic_pci_address"
2) "0000:09:00.0"
3) "module_name"
4) "FABRIC-CARD0"
5) "asic_id_in_module"
6) "0"

Reply:
Like we discussed in the meeting, maximum asic number will be discussed in next phrase.